lens-cube/src/main/java/org/apache/lens/cube/parse/StorageCandidate.java

```diff
@@ -518,13 +575,44 @@ private boolean evaluateMeasuresCompleteness(TimeRange timeRange) throws LensExc
   }
 
   @Override
-  public Set<FactPartition> getParticipatingPartitions() {
-    return null;
+  public boolean isExpressionEvaluable(ExpressionResolver.ExpressionContext expr) {
+    return expr.isEvaluable(this);
   }
 
   @Override
-  public boolean isExpressionEvaluable(ExpressionResolver.ExpressionContext expr) {
-    return expr.isEvaluable(this);
+  public void updateAnswerableSelectColumns(CubeQueryContext cubeql) throws LensException {
+    // update select AST with selected fields
+    int currentChild = 0;
+    for (int i = 0; i < cubeql.getSelectAST().getChildCount(); i++) {
+      ASTNode selectExpr = (ASTNode) queryAst.getSelectAST().getChild(currentChild);
```

Comment:
I think we can keep all the measures here. Later in Union query rewrite we are setting the non answerable ones with default "0.0" to make sure count and order of projected columns are same.